src/main/java/com/iexec/core/task/update/TaskUpdatePriorityBlockingQueue.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.core.task.update;
+
+import com.iexec.core.task.Task;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.stream.Stream;
+
+public class TaskUpdatePriorityBlockingQueue extends PriorityBlockingQueue<Runnable> {
+    /**
+     * Inserts the specified element into this priority queue.
+     * As the queue is unbounded, this method will never return {@code false}.
+     * <br>
+     * However, if the parameter is not a {@link TaskUpdate} element,
+     * it will throw an {@link UnsupportedOperationException}.
+     *
+     * @param runnable The element to add
+     * @return {@code true} (as specified by {@link Collection#add})
+     */
+    @Override
+    public boolean add(Runnable runnable) {
+        return this.offer(runnable);
+    }
+
+    /**
+     * Inserts the specified element into this priority queue.
+     * As the queue is unbounded, this method will never return {@code false}.
+     * <br>
+     * However, if the parameter is not a {@link TaskUpdate},
+     * it will throw an {@link UnsupportedOperationException}.
+     *
+     * @param runnable The element to add
+     * @return {@code true} (as specified by {@link Queue#offer})
+     */
+    @Override
+    public boolean offer(Runnable runnable) {
+        if (!(runnable instanceof TaskUpdate)) {
+            throw new UnsupportedOperationException("TaskUpdatePriorityBlockingQueue can only have TaskUpdate elements." +
+                    " [addedElementClass: " + runnable.getClass().getSimpleName() + "]");
+        }
+        return super.offer(runnable);
+    }
+
+    @Override
+    public TaskUpdate take() throws InterruptedException {
+        return (TaskUpdate) super.take();
+    }
+
+    /**
+     * Takes all the elements from this queue
+     * and returns them in a sorted {@code List} according to their priority.
+     *
+     * @return A {@link List} containing all the elements of the queue,
+     * sorted by priority.
+     * @throws InterruptedException if interrupted while waiting
+     */
+    public List<TaskUpdate> takeAll() throws InterruptedException {
+        List<TaskUpdate> elements = new ArrayList<>();
+        while (!isEmpty()) {
+            elements.add(take());
+        }
+        return elements;
+    }
+
+    /**
+     * Returns a {@link Stream} over this collection,
+     * whose elements are already cast to {@link TaskUpdate}.
+     *
+     * @return A {@link Stream} as described above.
+     */
+    public Stream<TaskUpdate> streamAsTaskUpdate() {
```

Comment:
streamOfTaskUpdate()
?

Reply:
I don't really like `of`. The idea here is to tell that the elements in the queue are streamed as `TaskUpdate`s. I don't think `streamOfTaskUpdate` carries the same meaning: it tells this method returns a stream of `TaskUpdate`s but does not say anything about where these objects come from.